github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/16040464?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/16040464?buildTab=artifacts#meta) on refs/heads/master @ [e762e056b97a](https://github.com/cockroachdb/pebble/commits/e762e056b97a):


```
        // 09:02:13.180 INFO: [JOB 276] compacting: sstable created 025071
        // 09:02:13.189 INFO: [JOB 275] compacted(download,copy) L0 [022161] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025070] (812B), in 0.0s (2.7s total), output rate 265KB/s
        // 09:02:15.944 INFO: [JOB 278] compacting(download,copy) L0 [023089] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:15.960 INFO: [JOB 276] compacted(default) L0 [011448 011724 012246 013633 014211 014842 015627 015824 016793 017191 017478 017658 018047 018667 018952 019213 019550 020322 021016 021842 022453 022901 023427 023897] (21KB) Score=26566.03 + L6 [019711] (3.0KB) Score=0.01 -> L6 [025071] (1.7KB), in 0.0s (2.8s total), output rate 188KB/s
        // 09:02:18.464 INFO: [JOB 279] compacting(default) L0 [016727 017124 017412 017591 017740 017978 018599 018885 019145 019480 019769 020066 020255 020628 020948 021499 021768 021905 022099 022821 023348 023817 024332 024705] (21KB) Score=26743.74 + L6 [025053] (978B) Score=0.01; OverlappingRatio: Single 0.05, Multi 0.00
        // 09:02:18.473 INFO: [JOB 279] compacting: sstable created 025073
        // 09:02:18.480 INFO: [JOB 278] compacted(download,copy) L0 [023089] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025072] (812B), in 0.0s (2.5s total), output rate 214KB/s
        // 09:02:21.271 INFO: [JOB 281] compacting(download,copy) L0 [022620] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:21.285 INFO: [JOB 279] compacted(default) L0 [016727 017124 017412 017591 017740 017978 018599 018885 019145 019480 019769 020066 020255 020628 020948 021499 021768 021905 022099 022821 023348 023817 024332 024705] (21KB) Score=26743.74 + L6 [025053] (978B) Score=0.01 -> L6 [025073] (931B), in 0.0s (2.8s total), output rate 80KB/s
        // 09:02:23.985 INFO: [JOB 279] sstable deleted 024705
        // 09:02:23.985 INFO: [JOB 282] compacting(default) L0 [011447 011723 012245 013632 014210 014841 015626 015823 016792 017190 017477 017657 018046 018666 018951 019212 019549 020321 021015 021327 021840 021841 021979 022451 022452 022898 022899 022900 023425 023426 023896] (27KB) Score=26649.63 + L6 [025071] (1.7KB) Score=0.01; OverlappingRatio: Single 0.06, Multi 0.00
        // 09:02:23.986 INFO: [JOB 279] sstable deleted 025053
        // 09:02:23.994 INFO: [JOB 282] compacting: sstable created 025075
        // 09:02:24.003 INFO: [JOB 281] compacted(download,copy) L0 [022620] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025074] (812B), in 0.0s (2.7s total), output rate 272KB/s
        // 09:02:26.695 INFO: [JOB 284] compacting(download,copy) L0 [021337] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:26.696 INFO: [JOB 285] compacting(download,copy) L0 [022611] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:26.710 INFO: [JOB 282] compacted(default) L0 [011447 011723 012245 013632 014210 014841 015626 015823 016792 017190 017477 017657 018046 018666 018951 019212 019549 020321 021015 021327 021840 021841 021979 022451 022452 022898 022899 022900 023425 023426 023896] (27KB) Score=26649.63 + L6 [025071] (1.7KB) Score=0.01 -> L6 [025075] (1.8KB), in 0.0s (2.7s total), output rate 137KB/s
        // 09:02:29.266 INFO: [JOB 282] sstable deleted 025071
        // 09:02:29.266 INFO: [JOB 286] compacting(default) L0 [015610 015806 016776 017172 017460 017640 017789 018028 018648 018933 019194 019531 019819 020116 020306 020997 021549 021818 021957 022150 022870 023397 023867 024379] (21KB) Score=26497.01 + L6 [015377] (835B) Score=0.01; OverlappingRatio: Single 0.04, Multi 0.00
        // 09:02:29.280 INFO: [JOB 284] compacted(download,copy) L0 [021337] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025076] (812B), in 0.0s (2.6s total), output rate 256KB/s
        // 09:02:31.829 INFO: [JOB 288] compacting(download,copy) L0 [023228] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:31.844 INFO: [JOB 286] compacted(default) L0 [015610 015806 016776 017172 017460 017640 017789 018028 018648 018933 019194 019531 019819 020116 020306 020997 021549 021818 021957 022150 022870 023397 023867 024379] (21KB) Score=26497.01 + L6 [015377] (835B) Score=0.01 -> L6 [] (0B), in 0.0s (2.6s total), output rate 0B/s
        // 09:02:34.454 INFO: [JOB 289] compacting(default) L0 [015609 015804 016775 017171 017459 017639 017788 018027 018647 018932 019193 019530 019818 020115 020305 020996 021548 021817 021956 022149 022869 023396 023866 024378] (21KB) Score=26576.81 + L6 [] (0B) Score=0.01; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:34.470 INFO: [JOB 285] compacted(download,copy) L0 [022611] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025077] (812B), in 0.0s (7.8s total), output rate 265KB/s
        // 09:02:37.086 INFO: [JOB 291] compacting(download,copy) L0 [021080] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:37.086 INFO: [JOB 292] compacting(download,copy) L0 [021857] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:37.101 INFO: [JOB 288] compacted(download,copy) L0 [023228] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025078] (812B), in 0.0s (5.3s total), output rate 268KB/s
        // 09:02:39.570 INFO: [JOB 294] compacting(download,copy) L0 [020179] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:39.584 INFO: [JOB 292] compacted(download,copy) L0 [021857] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025080] (812B), in 0.0s (2.5s total), output rate 264KB/s
        // 09:02:42.072 INFO: [JOB 296] compacting(download,copy) L0 [018326] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:42.086 INFO: [JOB 294] compacted(download,copy) L0 [020179] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025081] (812B), in 0.0s (2.5s total), output rate 251KB/s
        // 09:02:44.604 INFO: [JOB 298] compacting(download,copy) L0 [024427] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:44.619 INFO: [JOB 296] compacted(download,copy) L0 [018326] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025082] (812B), in 0.0s (2.5s total), output rate 258KB/s
        // 09:02:47.072 INFO: [JOB 300] compacting(download,copy) L0 [022614] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:47.087 INFO: [JOB 298] compacted(download,copy) L0 [024427] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025083] (812B), in 0.0s (2.5s total), output rate 287KB/s
        // 09:02:49.646 INFO: [JOB 302] compacting(download,copy) L0 [022616] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:49.661 INFO: [JOB 300] compacted(download,copy) L0 [022614] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025084] (812B), in 0.0s (2.6s total), output rate 254KB/s
        // 09:02:52.108 INFO: [JOB 304] compacting(download,copy) L0 [018182] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:52.123 INFO: [JOB 302] compacted(download,copy) L0 [022616] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025085] (812B), in 0.0s (2.5s total), output rate 196KB/s
        // 09:02:54.570 INFO: [JOB 306] compacting(download,copy) L0 [023222] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:54.585 INFO: [JOB 304] compacted(download,copy) L0 [018182] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025086] (812B), in 0.0s (2.5s total), output rate 221KB/s
        // 09:02:57.121 INFO: [JOB 308] compacting(download,copy) L0 [024759] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:57.136 INFO: [JOB 306] compacted(download,copy) L0 [023222] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025087] (812B), in 0.0s (2.6s total), output rate 250KB/s
        // 09:02:59.609 INFO: [JOB 310] compacting(download,copy) L0 [020187] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:02:59.624 INFO: [JOB 308] compacted(download,copy) L0 [024759] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025088] (812B), in 0.0s (2.5s total), output rate 268KB/s
        // 09:03:02.090 INFO: [JOB 312] compacting(download,copy) L0 [021988] (867B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 09:03:02.105 INFO: [JOB 310] compacted(download,copy) L0 [020187] (867B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [025089] (812B), in 0.0s (2.5s total), output rate 267KB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/240712-085041.033486664335/random-017 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-017 (736.10s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1720774241033258527 -ops "uniform:5000-10000"
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #3681 github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed [A-storage C-test-failure O-metamorphic O-robot T-storage branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: PEBBLE-221

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/16088264?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/16088264?buildTab=artifacts#meta) on refs/heads/master @ [b6c49f44e8cf](https://github.com/cockroachdb/pebble/commits/b6c49f44e8cf):


```
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006382
        // 05:09:45.382 INFO: [JOB 2728] flushing: sstable created 010244
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006383
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006384
        // 05:09:45.382 INFO: [JOB 2728] flushing: sstable created 010245
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006389
        // 05:09:45.382 INFO: [JOB 2728] flushing: sstable created 010246
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006390
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006391
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006392
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006393
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006394
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006395
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006396
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006402
        // 05:09:45.382 INFO: [JOB 2696] sstable deleted 006403
        // 05:09:45.384 INFO: [JOB 2728] flushing: sstable created 010247
        // 05:09:45.386 INFO: [JOB 2734] compacted(default) L0 [006450 006739 006913 007107 007507 007633 007796 008019 008205 008475 008673 008976 009198 009450 009757] (11KB) Score=462.89 + L1 [] (0B) Score=0.00 -> L1 [010230] (941B), in 2.2s (2.2s total), output rate 419B/s
        // 05:09:45.389 INFO: [JOB 2696] sstable deleted 006404
        // 05:09:45.399 INFO: [JOB 2728] flushing: sstable created 010248
        // 05:09:48.275 INFO: [JOB 2736] compacting: sstable created 010249
        // 05:09:48.275 INFO: [JOB 2738] compacting(move) multilevelL1 [010230] (941B) Score=11762.50 + L2 [] (0B) Score=0.00 + L3 [] (0B) Score=2.40; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:09:48.275 INFO: [JOB 2728] flushing: sstable created 010250
        // 05:09:48.276 INFO: [JOB 2738] compacted(move) multilevelL1 [010230] (941B) Score=11762.50 + L2 [] (0B) Score=0.00 + L3 [] (0B) Score=2.40 -> L3 [010230] (941B), in 0.0s (0.0s total), output rate 158MB/s
        // 05:09:50.734 INFO: [JOB 2696] sstable deleted 006406
        // 05:09:50.734 INFO: [JOB 2739] compacting(default) L0 [004690 004489 004887 005359 005360 005361 005362 005644 005645 005646 005647 006023 006024 006025 006026 006363 006364 006365 006366 006649 006650 006651 006652 007308 007309 007310 007311 007398 007399 007400 007401 007683 007684 007685 007686 008357 008358 008359 008360 008845 008846 008847 008848 009069 009070 009071 009072 008790 009345 009899] (37KB) Score=459.96 + L1 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:09:50.734 INFO: [JOB 2696] sstable deleted 006407
        // 05:09:50.734 INFO: [JOB 2728] flushing: sstable created 010251
        // 05:09:50.734 INFO: [JOB 2696] sstable deleted 006408
        // 05:09:50.735 INFO: [JOB 2736] compacted(default) multilevelL3 [010155] (1.0KB) Score=3.28 + L4 [] (0B) Score=2.15 + L5 [] (0B) Score=0.66 -> L5 [010249] (1.0KB), in 2.9s (5.4s total), output rate 369B/s
        // 05:09:53.319 INFO: [JOB 2696] sstable deleted 006409
        // 05:09:53.320 INFO: [JOB 2741] compacting(default) multilevelL3 [010230] (941B) Score=3.17 + L4 [] (0B) Score=2.12 + L5 [] (0B) Score=0.67; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006410
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006411
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006412
        // 05:09:53.320 INFO: [JOB 2741] compacting: sstable created 010252
        // 05:09:53.320 INFO: [JOB 2739] compacting: sstable created 010253
        // 05:09:53.320 INFO: [JOB 2728] flushing: sstable created 010254
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006413
        // 05:09:53.320 INFO: [JOB 2728] flushing: sstable created 010255
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006414
        // 05:09:53.320 INFO: [JOB 2728] flushing: sstable created 010256
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006415
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006416
        // 05:09:53.320 INFO: [JOB 2696] sstable deleted 006417
        // 05:09:53.320 INFO: [JOB 2728] flushing: sstable created 010257
        // 05:09:53.321 INFO: [JOB 2741] compacted(default) multilevelL3 [010230] (941B) Score=3.17 + L4 [] (0B) Score=2.12 + L5 [] (0B) Score=0.67 -> L5 [010252] (941B), in 0.0s (0.0s total), output rate 1.7MB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/240716-045404.9842585609778/random-019 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-019 (947.82s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1721105644984393815 -ops "uniform:5000-10000"
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #3681 github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed [A-storage C-test-failure O-metamorphic O-robot T-storage branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/vfs/errorfs/latency.go
+++ b/vfs/errorfs/latency.go
@@ -27,6 +27,7 @@ func RandomLatency(pred Predicate, mean time.Duration, seed int64, limit time.Du
 	rl := &randomLatency{
 		predicate: pred,
 		mean:      mean,
+		limit:     limit,
 	}
 	rl.keyedPrng.init(seed)
 	return rl

